src/main/java/com/minecolonies/coremod/colony/buildings/BuildingBuilder.java

```diff
@@ -246,6 +352,60 @@ public void resetNeededResources()
     }
 
     /**
+     * Update the available resources (from the chest or builder's inventory)
+     * which are needed for the build
+     */
+    private void updateAvailableResources()
+    {
+        final EntityCitizen builder = getWorkerEntity();
+        resourcesAvailable.clear();
+
+
+        InventoryCitizen builderInventory = null;
+        if (builder!=null)
+        {
+            builderInventory = builder.getInventoryCitizen();
+        }
+
+
+        for (@NotNull final Map.Entry<String, ItemStack> entry : neededResources.entrySet())
+        {
+            ItemStack itemStack = entry.getValue();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Local variable 'itemStack' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)